frontend/ui/network/manager.lua

```diff
@@ -60,6 +61,39 @@ function NetworkMgr:promptWifiOff(complete_callback)
     })
 end
 
+function NetworkMgr:turnOnWifiAndWaitForConnection(callback)
```

Comment:
There doesn't currently seem to be a way to interrupt this? There's a clever method implemented by @poire-z in https://github.com/koreader/koreader/pull/3224

Reply:
It won't be that easy.
To use these Trapper methods (beware #3224 is a bit old, and stuff has been rewritten in #3228), you need to Trapper:wrap() some code well up above the call to NetworkMgr:turnOnWifiAndWaitForConnection() (because when we want to allow dismissal, we coroutine.yield(), and this exits to where we called Trapper:wrap()).
So we would need to Trapper:wrap() at the correct place all the actions that could make use of this method...
(Unless we rethink the whole thing to have all koreader in a coroutine ... or have UIManager wrap() as soon as is sends and Event... or stuff like that I haven't really think about :)

Reply:
Trapper seems to be a big change, too complicated. 
How about marking this as TODO for next PR?

Reply:
Just like Trapper was not from begging with Wikipedia (#3224) and Dictionary (#3228), it was added later on, with those commits.

Reply:
Neither of those are comparable. Dictionary taking a few seconds is long. Wikipedia download is impossible to semi-accidentally trigger. I suppose your point is that auto-wifi enable isn't the default. Would an extra warning message be sufficient?

Reply:
yes + 30seconds is not an ages. We already have info how long someone would have to wait
`T(_("Enabling Wi-Fi. Waiting for Internet connection…\nTimeout %1 seconds.")`
I think adding extra message in unnecessary. When do you want that message to be shown? 
when you select "Auto turn on wifi"  ? IMHO doesn't make sense.

Reply:
> We already have info how long someone would have to wait

User reaction: oh @#$@# I'm not at home cancel cancel cancel cancel (times 60). :-)

> when you select "Auto turn on wifi" ? IMHO doesn't make sense.

It occurs to me that we can capture the same information by calling the menu item `Auto turn on wifi (experimental)`

Reply:
Ok. Experimental seems to be a good compromise